src/drawtypes/label.cpp

```diff
@@ -148,14 +148,23 @@ namespace drawtypes {
     if (required) {
       text = conf.get(section, name);
     } else {
-      text = conf.get(section, name, move(def));
+      text = conf.get(section, name, def);
     }
 
-    const auto get_left_right = [&](string key) {
-      auto value = conf.get(section, key, 0U);
-      auto left = conf.get(section, key + "-left", value);
-      auto right = conf.get(section, key + "-right", value);
-      return side_values{static_cast<unsigned short int>(left), static_cast<unsigned short int>(right)};
+    const auto get_left_right = [&](string&& key) {
+      const auto parse_or_throw = [&](const string& key, space_size default_value) {
+        try {
+          return conf.get(section, key, default_value);
+        } catch (const std::exception& err) {
+          throw application_error(
```

Comment:
I really don't like that we have to handle malformed values in the places where they are accessed when we even use a `convert` function to parse them. There's not much we can do about that right now though :disappointed:. I'll probably takle this when we rework the convert functions in the new config parser

Reply:
I agree, a verification phase would be welcome